src/assemble_workflow/dist.py

```diff
@@ -36,42 +41,145 @@ def extract(self, dest):
+        os.makedirs(data_dir_path, exist_ok=True)
+        os.makedirs(log_dir_path, exist_ok=True)
+        shutil.copyfile("scripts/pkg/scripts/systemd-entrypoint", systemed_entrypoint_path)
+        subprocess.run(
+            [
+                "fpm",
+                "--force",
+                "--verbose",
+                "--input-type",
+                "dir",
+                "--package",
+                os.path.join(root, bundle_recorder.package_name),
+                "--output-type",
+                self.get_distribution,
+                "--name",
+                product_name,
+                "--description",
+                " ".join([product_name , self.get_distribution, manifest_data["version"]]),
+                "--version",
+                manifest_data["version"],
+                "--url",
+                "https://opensearch.org/",
+                "--vendor",
+                "OpenSearch",
+                "--maintainer",
+                "OpenSearch",
+                "--license",
+                "ASL 2.0",
+                "--before-install",
+                os.path.join(scripts_path, "scripts", "pre_install.sh"),
+                "--before-remove",
+                os.path.join(scripts_path, "scripts", "pre_remove.sh"),
+                "--after-install",
+                os.path.join(scripts_path, "scripts", "post_install.sh"),
+                "--after-remove",
+                os.path.join(scripts_path, "scripts", "post_remove.sh"),
+                "--config-files",
+                os.path.join(os.sep, "etc", product_name, product_name_alt + ".yml"),
+                "--template-value",
+                "product=" + product_name,
+                "--template-value",
+                "user=" + product_name,
+                "--template-value",
+                "group=" + product_name,
+                "--template-value",
+                "homeDir=" + os.path.join(os.sep, "usr", "share", product_name),
+                "--template-value",
+                "configDir=" + os.path.join(os.sep, "etc", product_name),
+                "--template-value",
+                "pluginsDir=" + os.path.join(os.sep, "usr", "share", product_name, "plugins"),
+                "--template-value",
+                "dataDir=" + os.path.join(os.sep, "var", "lib", product_name),
+                "--template-value",
+                "logDir=" + os.path.join(os.sep, "var", "log", product_name),               
+                "--exclude",
+                os.path.join("usr", "share", product_name, "data"),
+                "--exclude",
+                os.path.join("usr", "share", product_name, "config"),
+                "--architecture",
+                architecture_alt,
+                os.path.join(real_archive_path, "") + "=" + os.path.join(os.sep, "usr", "share", product_name, ""),
+                os.path.join(real_archive_path, "config", "") + "=" + os.path.join(os.sep, "etc", product_name, ""),
```

Comment:
why is there an empty "" here? Is it to add a `/`?

Reply:
Yes, in order to add "/" at the end.

Reply:
Do these parameters refer to actual paths, settings or something else? Do they have to be *nix paths always? 

If yes, then you don't want `os.path.join` because on Windows this will result in `\` instead of `/`.

Reply:
Yes, they are actual paths, e.g. `usr/share/opensearch/` Does it exactly what we want here? using `\` as seperator in Windows and `/` in *nix?